internal/lang/funcs/cidr_test.go

```diff
@@ -396,3 +396,107 @@ func TestCidrSubnets(t *testing.T) {
 		})
 	}
 }
+
+func TestCidrContains(t *testing.T) {
+	noError := func(err error) bool { return err == nil }
+
+	tests := []struct {
+		Prefix  cty.Value
+		Address cty.Value
+		Want    cty.Value
+		ErrFn   func(error) bool
+	}{
+		{
+			// IPv4, contained (IP).
+			cty.StringVal("192.168.2.0/20"),
+			cty.StringVal("192.168.2.1"),
+			cty.True,
+			noError,
+		},
+		{
+			// IPv4, contained (CIDR).
+			cty.StringVal("192.168.2.0/20"),
+			cty.StringVal("192.168.2.0/22"),
+			cty.True,
+			noError,
+		},
+		{
+			// IPv4, not contained.
+			cty.StringVal("192.168.2.0/20"),
+			cty.StringVal("192.126.2.1"),
+			cty.False,
+			noError,
+		},
+		{
+			// IPv4, not contained (CIDR).
+			cty.StringVal("192.168.2.0/20"),
+			cty.StringVal("192.126.2.0/18"),
+			cty.False,
+			noError,
+		},
+		{
+			// IPv6, contained.
+			cty.StringVal("fe80::/48"),
+			cty.StringVal("fe80::1"),
+			cty.True,
+			noError,
+		},
+		{
+			// IPv6, not contained.
+			cty.StringVal("fe80::/48"),
+			cty.StringVal("fe81::1"),
+			cty.False,
+			noError,
+		},
+		{
+			// Address family mismatch: IPv4 prefix, IPv6 address.
```

Comment:
This tests ipv4 prefix and single ipv6 address.  We don't test ipv4 prefix with ipv6 prefix.

In the current implementation, we'll not get to the ipv4_prefix.contains(endIP) because of the ordering of the code.  However, maybe someone refactors the code in the future?  But here, we're really testing the internal ipaddr module, so if it does the right thing we might be find.

Reply:
Done.